delegation-verify-coordinator/templates/rolebinding_job.yaml

```diff
@@ -2,13 +2,13 @@
 apiVersion: rbac.authorization.k8s.io/v1
 kind: RoleBinding
 metadata:
-  name: {{ .Release.Name }}
+  name: {{ .Release.Name }}-job
```

Comment:
the `-job` suffix, is confusing as this role binding is applied to a kind `deployment`, not a kind `job`